### DISCUSSION TITLE
How do I select from related table in reverse way?

### DISCUSSION THREAD

**Discussion by MML1357** [prisma/prisma#7759] [Q&A] [answered]
Hi. If I have this example models:

mode event {
  id                    Int         @id @default(autoincrement())
  name              String?     @db.VarChar
  active              Boolean     @default(true)
  area_id           String?     @db.VarChar
}

model area {
  id                    Int         @id @default(autoincrement())
  name              String?     @db.VarChar
  event              event[]
}

There's a way to make a prisma.event.findMany to get all the events and the id and name of the related area?
I know it's suppose to work in the other way: prisma.area.findMany

**Comment by harbar20** [comment]
Can you get your records in an array, then `arr.reverse()` it?

**Reply by MML1357** [reply]
Hi. That's not what I'm looking for.

**Comment by ryands17** [comment, marked as answer]
@MML1357 👋 

You have modelled it incorrectly. This schema will not work as Prisma always provides a back relation to the model in a 1-many relation. The correct way to do this is:

```prisma
model event {
  id     Int     @id @default(autoincrement())
  name   String? @db.VarChar
  active Boolean @default(true)
  area   area?   @relation(fields: [areaId], references: [id])
  areaId Int?
}

model area {
  id    Int     @id @default(autoincrement())
  name  String? @db.VarChar
  event event[]
}
```

Then you can find the area as follows:

```ts
await prisma.event.findMany({ include: { area: true } })
```

You can read more about how to model relations and query for them here:
https://www.prisma.io/docs/concepts/components/prisma-schema/relations
https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries

**Reply by MML1357** [reply]
Hi. Thanks for answering.
Can I do that working without affecting the real database? I mean just by modifying the model in the prisma.schema file.

**Reply by ryands17** [reply]
Yes that should work. Also saving `schema.prisma` should automatically add the relation if you have this plugin installed with VSCode: https://marketplace.visualstudio.com/items?itemName=Prisma.prisma

**Reply by MML1357** [reply]
@ryands17, that works like a charm.So, let me take advantage and make another question.
Having
`const event = await prisma.event.findMany({ include: { area: true } })`
What if I want `event` has an extra field called `state`, which value should be `true` if an `event id` is stored in `area`, and `false` if don't.
Can I make it possible without iterating over `event` and assigning manually?

**Reply by ryands17** [reply]
Depends, if that field is in your mode, then you can set it to `true` when the `event id` is mapped to an area and false if not mapped so you do not need to perform this while fetching.

If that field is not in the model, then you would need to do it while fetching.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
